Python/bytecodes.c

```diff
@@ -2879,13 +2879,13 @@ dummy_func(
             exc_info->exc_value = Py_NewRef(new_exc);
         }
 
-        op(_GUARD_DORV_VALUES_INST_ATTR_FROM_DICT, (owner -- owner)) {
+        guard op(_GUARD_DORV_VALUES_INST_ATTR_FROM_DICT, (owner -- owner: ~(GUARD_DORV_VALUES_INST_ATTR_FROM_DICT_TYPE))) {
```

Comment:
What does `GUARD_DORV_VALUES_INST_ATTR_FROM_DICT_TYPE` mean?
Is this just saying that a value guarded by `_GUARD_DORV_VALUES_INST_ATTR_FROM_DICT` has been guarded by `_GUARD_DORV_VALUES_INST_ATTR_FROM_DICT`? That seems redundant.

Reply:
All guards add information, whether that is type information depends on the type system being used.

This implies that your type system includes whether an object has a `__dict__` or not, but that's not part of the what most people would consider the Python type system.

A type system that tracks stack space is plausible. In which case `_CHECK_STACK_SPACE` would carry type information.